src/vs/workbench/api/common/menusExtensionPoint.ts

```diff
@@ -176,6 +176,11 @@ const apiMenus: IAPIMenu[] = [
 		id: MenuId.TimelineItemContext,
 		description: localize('view.timelineContext', "The Timeline view item context menu")
 	},
+	{
+		key: 'search/context',
+		id: MenuId.SearchContext,
+		description: localize('menus.searchContext', "The search context menu")
+	},
```

Comment:
@roblourens This smell like "under the radar" API work. Esp. interesting is what arguments those commands are invoked with. We have no API representation for search results and shouldn't introduce one with these changes unless properly reviewed and designed

Reply:
I thought the whole space of arguments for commands is sort of under the radar API. I looked at some other context menu contexts and they are invoked with different arbitrary things.

Reply:
This is API since changing arguments breaks extensions. I would be surprised if context menus use arbitrary objects as arguments, the rule should be use URI whenever the source itself has no representation in the API, eg. the explorer uses URI, so does the editor title and editor context, but SCM uses SCM things, custom tree use TreeItem etc.

For search results a URI isn't enough and search API (which I assume would talk about search matches) has never been tackled for finalization

Reply:
I was looking at variables and callstacks, which as far as I can tell are arbitrary objects with no api representation @isidorn.

Reply:
@roblourens correct, they have no official vscode api representation.
Here's a expert from the documentation for more details:

"When a registered context menu command is executed, both the underlying variable and its container are passed as Debug Adapter Protocol (DAP) objects. Please note that VS Code's extension API uses opaque stand-in types instead of the real DAP types. In order to access their properties, they can be easily coerced into the corresponding DAP types."

Reply:
On the callstack, I saw this 
![image](https://user-images.githubusercontent.com/323878/98887442-db476080-244a-11eb-9009-af0a0b36952b.png) which I think isn't associated with any single DAP type, but I guess it's all coming directly from stuff in the DAP.

Let's reopen the issue for discussion. I'm looking at the proposed API that we have for search results and trying to decide whether it makes sense to use the same objects here. You could make a case either way. If we can make a case to design this api without necessarily tying it to search provider API, then we can do this without making it part of the search provider api problem.

One difference between the UI results and the provider results is that providers return one Match per line, possibly with multiple ranges on the line. The UI object always has one object per actual match.

Reply:
Yeah unlike the variables view the callstack is in a bad shape and I have this issue to tackle it https://github.com/microsoft/vscode/issues/85696
We can continue the discussion there